src/beach.ts

```diff
@@ -0,0 +1,47 @@
+import { cliExecute, fileToBuffer, gamedayToInt, myId } from "kolmafia";
+import { get, set } from "libram";
+
+type BeachTile = { minute: number; row: number; column: number };
+
+const rareTiles: BeachTile[] = JSON.parse(fileToBuffer("raretiles.json"));
+
+function _comb(tile: BeachTile): void {
+  const { minute, row, column } = tile;
+  cliExecute(`beach wander ${minute}; beach comb ${row} ${column}`);
+}
+
+let seed = parseInt(myId());
+function deterministicRandom(): number {
+  seed++;
+  return Math.sin(seed);
+}
+
+function deterministicShuffle<T>(array: T[]): T[] {
+  const returnValue: T[] = [];
+  while (array.length > 0) {
+    const index = Math.floor(deterministicRandom() * array.length);
+    returnValue.push(...array.splice(index));
+  }
+  return returnValue;
+}
+
+let shuffledArray: BeachTile[];
+
+function getShuffledArray(): BeachTile[] {
+  if (!shuffledArray) {
+    shuffledArray = deterministicShuffle(rareTiles);
+  }
+  return shuffledArray;
+}
+
+export default function comb(): void {
+  const tileList = getShuffledArray();
+  const index = (get("garbo_lastTileCombed", 0) + 1) % tileList.length;
+  const tile = tileList[index];
+  const dayOfMonth = 1 + (gamedayToInt() % 8);
+  const rowsHidden = 4 - Math.abs(4 - dayOfMonth);
+  const shouldComb = tile.row > rowsHidden;
+  if (shouldComb) _comb(tile);
+  set("garbo_lastTileCombed", index);
```

Comment:
What I'm reading here is that if the tide is high, we will skip tiles and act as if we had hit them and keep moving. This is likely fine for a first pass. On a second pass, we might want to prioritize tidal tiles in some way bc you can't hit them every day and we don't want them to go to "waste". (I really like this problem bc it has so many constraints)

Reply:
That's a very good point--I'm not sure how to do that in this paradigm, but I think you're also right that we should just push this without addressing it, and address it later.